macros/models/_project/yield_yak/yield_yak_reinvests.sql

```diff
@@ -0,0 +1,133 @@
+{% macro get_apy_from_array() %}
+    CREATE OR REPLACE FUNCTION get_apy_from_array(recent_reinvest_info ARRAY(varchar))
```

Comment:
Maybe you cannot create UDF here.

Reply:
yes, that is correct, we cannot create functions on dunesql engine directly. dbt macros can only be used to output sql code to run

Reply:
Okay thanks both, a couple of follow ups:

1. What about if the outputted SQL code includes an inline function ("WITH FUNCTION get_apy_from_array..")? In this format, the compile code created by "dbt compile" works without a problem but when I tried to add this to the PR it failed saying something like "inline functions cannot be used in views" (can see this in the previous commit). Is there any way around this? I thought I was creating incremental models, which I thought were materialized as tables rather than views (?) so I'm a bit confused by the error message.
2. What about the functions in the macros/public folder? Isn't there something I can do similarly to those to make this function work? i.e. defining it in a separate place/whatever is needed.

Reply:
Have realised that the historical information from the checks carried out is not visible so let me know if it would be beneficial for me to reverse the latest commit so that we're back to the inline function approach and you can see the "inline functions not allowed in views" error message when carrying out "dbt slim ci / dbt-test (spellbook)" checks

Reply:
I found the exact error message from previously by the way:
`Database Error in model yield_yak_arbitrum_reinvests (models/yield_yak/arbitrum/yield_yak_arbitrum_reinvests.sql)
TrinoUserError(type=USER_ERROR, name=NOT_SUPPORTED, message="line 10:1: Views cannot contain inline functions", query_id=20240415_171629_00071_v5q22)`

Reply:
Actually, tbh sorry about this guys. I've realised I'm not really making much use of routine-specific functionality in my function declaration for `get_apy_from_array`, so I think I can just put the logic directly in the SQL script and it should work (albeit with slightly worse readability - will add comments).